cpp/src/arrow/engine/substrait/extension_set.cc

```diff
@@ -1113,9 +1113,11 @@ struct DefaultExtensionIdRegistry : ExtensionIdRegistryImpl {
                                            DecodeBasicAggregate("mean")));
     DCHECK_OK(AddSubstraitAggregateToArrow({kSubstraitArithmeticFunctionsUri, "std_dev"},
                                            DecodeBasicAggregate("stddev")));
-    DCHECK_OK(
-        AddSubstraitAggregateToArrow({kSubstraitAggregateGenericFunctionsUri, "count"},
-                                     DecodeBasicAggregate("count")));
+    for (const auto& fn_name : {"count", "first", "last"}) {
+      DCHECK_OK(
+          AddSubstraitAggregateToArrow({kSubstraitAggregateGenericFunctionsUri, fn_name},
```

Comment:
Should we uri the arrow specifc uri for first/last instead of the substrait one?

Reply:
Right. Fixed.